ballerina/yaml_api.bal

```diff
@@ -16,16 +16,49 @@
 
 import ballerina/jballerina.java;
 
-# Converts YAML string, byte[] or byte-block-stream to subtype of anydata.
+# Converts YAML string to subtype of anydata.
 #
-# + s - Source YAML string value or byte[] or byte-block-stream
+# + s - Source string value
 # + options - Options to be used for filtering in the projection
 # + t - Target type
 # + return - On success, returns the given target type value, else returns an `yaml:Error`
-public isolated function fromYamlStringWithType(string|byte[]|stream<byte[], error?> s,
+public isolated function parseString(string s,
         Options options = {}, typedesc<anydata> t = <>)
-        returns t|Error = @java:Method {'class: "io.ballerina.stdlib.data.yaml.Native"} external;
-        
+    returns t|Error = @java:Method {'class: "io.ballerina.lib.data.yaml.Native"} external;
+
+# Converts YAML byte[] to subtype of anydata.
+#
+# + s - Source byte[] value
+# + options - Options to be used for filtering in the projection
+# + t - Target type
+# + return - On success, returns the given target type value, else returns an `yaml:Error`
+public isolated function parseBytes(byte[] s,
+        Options options = {}, typedesc<anydata> t = <>)
+    returns t|Error = @java:Method {'class: "io.ballerina.lib.data.yaml.Native"} external;
+
+# Converts YAML byte-block-stream to subtype of anydata.
+#
+# + s - Source byte-block-stream value
+# + options - Options to be used for filtering in the projection
+# + t - Target type
+# + return - On success, returns the given target type value, else returns an `yaml:Error`
+public isolated function parseStream(stream<byte[], error?> s,
+        Options options = {}, typedesc<anydata> t = <>)
+    returns t|Error = @java:Method {'class: "io.ballerina.lib.data.yaml.Native"} external;
+
+# Converts anydata YAML value to a string.
+#
+# + yamlValue - Input yaml value
+# + config - Options used to get desired toString representation
+# + return - On success, returns to string value, else returns an `yaml:Error`
+public isolated function toYamlString(anydata yamlValue, WriteConfig config = {}) returns string|Error {
+    string[] lines = check toYamlStringArray(yamlValue, config);
+    return "\n".'join(...lines);
```

Comment:
is this correct for windows?

Reply:
Yes, since we are not writing this content to a file having this is fine.